apiserver/facades/agent/uniter/uniter.go

```diff
@@ -442,6 +443,13 @@ func (u *UniterAPI) Destroy(ctx context.Context, args params.Entities) (params.E
 				unit    *state.Unit
 				removed bool
 			)
+			err = u.unitRemover.DestroyUnit(ctx, tag.Id())
+			if err != nil && !errors.Is(err, applicationerrors.UnitNotFound) {
+				result.Results[i].Error = apiservererrors.ServerError(err)
+				continue
+			}
+
+			// TODO(units) - remove dual write to state
```

Comment:
I want to keep the `TODO(units)` so these are easily greppable as things pertaining to the units work.
There's currently no single jira card - the individual places where dual write is needed will be addressed as part of other work which is spread across various cards. I don't anticipate there will be a single PR which just removes them all in one go.
I created a new card https://warthogs.atlassian.net/browse/JUJU-6617 for the removal of unit logic from state but this will like evolve into a few cards since some aspects like cloud container can likely be done individually. It all depends on how the complex logic behind all this gets unpicked.

Reply:
We currently have 14 TODO(units), for various things, all of which will be knocked off as various cards are done. Having them makes working on the cards easier.